### ISSUE TITLE
(servicecatalogappregistry): validate that a Stack can only be associated with a single Application

### ISSUE THREAD

**Issue by arcrank** [aws/aws-cdk#15372]
Currently in new PR we are adding capabilities for associating resources to an AppRegistry application.  Seeing as a stack resource can only be associated with one application, it would be helpful if we added a background tracker to monitor all defined stacks in scope and fail fast (e.g. during `cdk synth`) if an end user tries to make two associations.  While this can't cover all cases (a user can import stacks, or declare them in a different app), it would help for cases that are in scope of a single app. 

https://github.com/aws/aws-cdk/pull/15371

### Use Case

Prevent late failures at deploy time that we can catch during synth

### Proposed Solution

Basically we will add an abstract class that will track all stacks added and then a mapping to an application if they have already been adding to one, and provide a user error that tells the user what app stack has already been added to, EXCEPT if they are adding it to the same application twice, which we treat as an idempotent request and ignore. 

### Other

* [x] :wave: I may be able to implement this feature request
* [ ] :warning: This feature might incur a breaking change

---

This is a :rocket: Feature Request
